### PR TITLE
WIP: base async step implementation

### PR DESCRIPTION
**Pull request by camillobruni**
This WIP PR adds `AsyncBenchmarkTestStep` and completely separate async run paths so the default runs are not affected.

### CODE REVIEW

File: resources/tests.mjs [hunk 1]
```diff
@@ -953,7 +953,27 @@ Suites.push({
...
+            page.querySelector("#prepare").click();
+            page.querySelector("#add-scatter-chart-button").click();
+        }),
+        new AsyncBenchmarkTestStep("Show tooltip", (page) => {
+            page.querySelector("#open-tooltip").click();
+        }),
+        new AsyncBenchmarkTestStep("Draw opaque scatter", (page) => {
+            page.querySelector("#opaque-color").click();
+            page.querySelector("#add-scatter-chart-button").click();
+        }),
```
Comment (smaug----) [inline resources/tests.mjs:974]: Hmm, what is async here? Should the test functions be async?

File: resources/benchmark-runner.mjs [hunk 2]
```diff
@@ -311,6 +337,23 @@ class RAFTestInvoker extends TestInvoker {
     }
 }
 
+class AsyncRAFTestInvoker extends BaseRAFTestInvoker {
+    _scheduleCallbacks(resolve) {
+        requestAnimationFrame(async () => {
+            await this._syncCallback();
```
Comment (smaug----) [inline resources/benchmark-runner.mjs:343]: Ok, so this makes sync callback the async thingie. Which makes sense given how callbacks are used currently, but naming is a bit confusing then. Perhaps we could rename the callbacks to not be sync and async, but something else. Not sure what.

Comment (camillobruni) [reply resources/benchmark-runner.mjs:343]: yeah absolutely, this is more confusing than ever :). 
- `main` vs .`trailing` or so?
- `blocking` vs. `delayed`?

File: resources/benchmark-runner.mjs [hunk 3]
```diff
@@ -311,6 +337,23 @@ class RAFTestInvoker extends TestInvoker {
...
+            requestAnimationFrame(() => {
```
Comment (smaug----) [inline resources/benchmark-runner.mjs:344]: This rAF is possibly a bit problematic. It is likely that after resolving syncCallback, one may need to wait quite a while before getting vsync, at least if the callback was very fast.
Another option could be to just immediately call setTimeout() => { this._asyncCallback(), but then that has similar issues what SP2 had where a paint might or might not be included in the measured time. So guess we do need rAF  here somehow.

Or, what if we run this._asyncCallback(); using queueMicrotask. That way promise callbacks triggered by _syncCallback() would get run before asyncCallback, but possible setTimeout(0)s wouldn't be.

File: resources/benchmark-runner.mjs (hunk 2 above)
Comment (flashdesignory) [reply resources/benchmark-runner.mjs:343]: How about something like this?

class TestInvoker {
    constructor(captureTest, captureLayout, reportResults) {
        this._captureTest = captureTest;
        this._captureLayout = captureLayout;
        this._reportResults = reportResults;
    }
}

**Pull request closed without merging**